File: edu/lagcc/opencc/models/request.py

class Request:

    def __init__(self, user, term, subject, class_num_5_digit, request_id=None):
        self.request_id = request_id
        self.user = user
        self.term = term
        self.subject = subject
        self.class_num_5_digit = class_num_5_digit

    def __str__(self):
        return "Request{request_id: "+str(self.request_id)+", user: "+self.user.__str__() + \
               ", term: "+self.term.__str__()+", subject: "+self.subject.__str__() + \
               ", class_num_5_digit: "+str(self.class_num_5_digit)+"}"

File: static/script/request.js
import * as modal from "./modal.js";

const add_request_form = document.querySelector('#add-request');
let formData, requestInfo;
add_request_form.onsubmit = evt => {
  evt.preventDefault();
  formData = new FormData(add_request_form);
  requestInfo = {
    "phone-number" : formData.get('phone-number'),
    "term" : formData.get('term'),
    "subject" : formData.get('subject'),
    "class-num-5" : formData.get('class-num-5'),
  }
  fetch('/api/request', {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(requestInfo)
  })
  .then(response => response.json())
  .then(data => 
    {
      if('success' in data){
        modal.showModal('#sucess-modal', data[Object.keys(data)[0]]);
        add_request_form.reset();
      }
      else
        modal.showModal('#error-modal', data[Object.keys(data)[0]])
    }
  );
}